lib/client.js

```diff
@@ -355,20 +356,36 @@ extend(Raven.prototype, {
   },
 
   captureException: function captureException(err, kwargs, cb) {
-    if (!(err instanceof Error)) {
-      // This handles when someone does:
-      //   throw "something awesome";
-      // We synthesize an Error here so we can extract a (rough) stack trace.
-      err = new Error(err);
-    }
-
     if (!cb && typeof kwargs === 'function') {
       cb = kwargs;
       kwargs = {};
     } else {
       kwargs = kwargs || {};
     }
 
+    if (!(err instanceof Error)) {
+      if (utils.isPlainObject(err)) {
+        // This will allow us to group events based on top-level keys
+        // which is much better than creating new group when any key/value change
```

Comment:
Damn, that's a great idea.

Reply:
![image](https://user-images.githubusercontent.com/1523305/34872448-264d04ac-f791-11e7-852e-1750b591ad5e.png)

Reply:
ah, that's awesome kamil. do they need to be hashed? I guess it's a length thing, right?

Reply:
Without hashing, very large objects will create way too large fingerprints. This way it's always fixed length as you mentioned.